tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/ChangelogUpdateTool.cs

```diff
@@ -0,0 +1,197 @@
+            CancellationToken ct)
+        {
+            // Prepare script parameters
+            var scriptParameters = new Dictionary<string, string>
+            {
+                { "SdkRepoPath", sdkRepoRoot },
+                { "PackagePath", packagePath }
+            };
+
+            // Create and execute process options for the update-changelog script
+            var processOptions = await CreateProcessOptions(ConfigType.UpdateChangelog, sdkRepoRoot, packagePath, scriptParameters, 15);
+            return await ExecuteProcessAsync(processOptions, ct, "Changelog content is updated.", ["Update the version if it's a release."]);
+        }
+
+        /// <summary>
+        /// Handles changelog guidance for data-plane packages.
+        /// </summary>
+        /// <returns>A noop response with guidance for manual editing.</returns>
+        private DefaultCommandResponse HandleDataPlaneChangelog()
+        {
+            logger.LogInformation("Data-plane package detected, providing manual editing guidance");
+            
+            return new DefaultCommandResponse
+            {
+                Result = "noop",
+                Message = "Data-plane changelog untouched; manual edits required.",
+                NextSteps = ["Update the version if it's a release."]
+            };
+        }
+
+        /// <summary>
+        /// Determines the SDK repository root from the package path.
+        /// </summary>
+        /// <param name="packagePath">The package path to start searching from.</param>
+        /// <returns>The SDK repository root path, or null if not found.</returns>
+        private string? DetermineSdkRepositoryRoot(string packagePath)
+        {
+            var currentPath = packagePath;
+            
+            // Look for common SDK repository indicators
+            var repoIndicators = new[] { ".git", "eng", "sdk", "packages" };
+            
+            while (!string.IsNullOrEmpty(currentPath) && currentPath != Path.GetPathRoot(currentPath))
+            {
+                if (repoIndicators.Any(indicator => Directory.Exists(Path.Combine(currentPath, indicator))))
+                {
+                    logger.LogInformation($"Detected SDK repository root at: {currentPath}");
+                    return currentPath;
+                }
+                
+                currentPath = Path.GetDirectoryName(currentPath);
+            }
+
+            logger.LogWarning($"Could not determine SDK repository root from package path: {packagePath}");
+            return null;
+        }
+
+        /// <summary>
+        /// Determines the package type (mgmt vs data-plane) for the specified package.
+        /// </summary>
+        /// <param name="packagePath">The package path.</param>
+        /// <param name="sdkRepoRoot">The SDK repository root.</param>
+        /// <returns>The package type string.</returns>
+        private string DeterminePackageType(string packagePath, string sdkRepoRoot)
```

Comment:
Same here.  It will be helpful to move this also to helper class if other tools need to know the SDK type